packages/theme/src/Components/TextField/index.tsx

```diff
@@ -88,6 +88,15 @@ export const MaskTextField = forwardRef((props: MaskTextFieldProps, ref: Forward
     const { label, sx, required = false, className, wrapperProps, helperText, ...rest } = props
     const InputProps = (props.InputProps as InputProps) ?? {}
     const { classes } = useStyles()
+    const onChange = useCallback(
+        (ev: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+            if (ev.currentTarget.value && !new RegExp(InputProps.inputProps?.pattern).test(ev.currentTarget.value)) {
+                return
+            }
+            props.onChange?.(ev)
+        },
+        [InputProps.inputProps?.pattern, props.onChange],
+    )
```

Comment:
this creates a mismatch between TextField and native HTML input. The pattern mismatch only blocks the submission of a form, it does not block the onChange event. I disagree with this change. cc @guanbinrui